File: Example Obj-c/Khalti-Pay/ViewController.m
//
//  ViewController.m
//  Khalti-Pay
//
//  Created by Rajendra Karki on 2/27/18.
//  Copyright © 2018 Rajendra Karki. All rights reserved.
//

#import "ViewController.h"
#import "Khalti-Swift.h"

@interface ViewController () <KhaltiPayDelegate>
    
    @property (nonatomic, strong) Config *config;

    @end

@implementation ViewController
    
    
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    
}
    
    
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

    
- (IBAction)pay:(UIButton *)sender {
    
    id objects[] = { @false, @true, @42, @12.23 };
    id keys[] = { @"no", @"yes", @"int", @"float" };
    NSUInteger count = sizeof(objects) / sizeof(id);
    NSDictionary *extra = [NSDictionary dictionaryWithObjects:objects
                                                      forKeys:keys
                                                        count:count];
    NSError *error;
    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:extra
                                                       options:(NSJSONWritingOptions)    (NSJSONWritingPrettyPrinted)
                                                         error:&error];
    
    
    NSString *jsonString;
    if (! jsonData) {
        NSLog(@"bv_jsonStringWithPrettyPrint: error: %@", error.localizedDescription);
        jsonString = @"{}";
    } else {
        jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
    }
    
    
    id extraObjects[] = { @"haha", jsonString };
    id extraKeys[] = { @"merchant_name", @"merchant_extra" };
    NSUInteger extraCount = sizeof(extraObjects) / sizeof(id);
    NSDictionary *additionalData = [NSDictionary dictionaryWithObjects:extraObjects
                                                      forKeys:extraKeys
                                                        count:extraCount];
    
    NSString * key = @"test_public_key_dc74e0fd57cb46cd93832aee0a507256";
    NSInteger *amount = 1000;
    Config *config = [[Config alloc] initWithPublicKey:key amount:amount productId:@"123123" productName:@"Dragon_boss" productUrl:@"as" additionalData:additionalData cardPayment: false];
    
    
    [Khalti presentWithCaller:self with:config delegate:self];
}

- (void)onCheckOutSuccessWithData:(NSDictionary<NSString *,id> *)data {
    NSLog(@"Oh there is Succes");
}

- (void)onCheckOutErrorWithAction:(NSString *)action message:(NSString *)message {
    NSLog(@"Oh there is error");
}

    @end

File: Example/Khalti/ViewController.swift
//
//  ViewController.swift
//  Khalti
//
//  Created by rjndra on 02/14/2018.
//  Copyright (c) 2018 rjndra. All rights reserved.
//

import UIKit
import Khalti

class ViewController: UIViewController, KhaltiPayDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func payNow(_ sender: UIButton) {
        let extra:[String : Any] =  ["no":false,"yes":true,"int" : 0, "float":12.23]
        
        let jsonData = try? JSONSerialization.data(withJSONObject: extra, options: JSONSerialization.WritingOptions())
        let jsonString = String(data: jsonData!, encoding: .utf8)!
        
        let additionalData:Dictionary<String,String> = [
            "merchant_name" : "haha",
            "merchant_extra" : jsonString
         ]
        
        Khalti.shared.appUrlScheme = khaltiUrlScheme
        let khaltiMerchantKey = "test_public_key_1d57e4d159794c6790a62f8953124053"
        
        let TEST_CONFIG:Config = Config(publicKey: khaltiMerchantKey, amount: 1000, productId: "1234567890", productName: "Dragon_boss", productUrl: "http://gameofthrones.wikia.com/wiki/Dragons",additionalData: additionalData, ebankingPayment: false)
        
        Khalti.shared.debugLog = true
        Khalti.present(caller: self, with: TEST_CONFIG, delegate: self)
    }
    
    func onCheckOutSuccess(data: Dictionary<String, Any>) {
        print(data)
        print("Oh there is success message received")
    }
    
    func onCheckOutError(action: String, message: String, data: Dictionary<String, Any>?) {
        print(action)
        print(message)
        if let value = data {
            print(value)
        }
        print("Oh there occure error in payment")
    }
    
}

